tmc-langs-ant/src/main/java/fi/helsinki/cs/tmc/langs/ant/AntPlugin.java

```diff
@@ -21,23 +32,164 @@ public String getLanguageName() {
-        return new File(path.toString() + File.separatorChar + "build.xml").exists();
+    /**
+     * Runs the build.xml file for the the given exercise.
+     *
+     * @param path The file path of the exercise directory.
+     * @return true if build success, else return false.
+     */
+    @VisibleForTesting
+    protected boolean buildAntProject(Path path) {
+        File buildFile = new File(path.toString() + File.separatorChar + "build.xml");
+        Project buildProject = new Project();
+        buildProject.setUserProperty("ant.file", buildFile.getAbsolutePath());
+        buildProject.setProperty("javac.fork", "true");
+        buildProject.init();
+        buildProject.setBaseDir(path.toAbsolutePath().toFile());
+        File buildLog;
+        
+        try {
+            
+            DefaultLogger logger = new DefaultLogger();
+            buildLog = new File(path.toString(), "build_log.txt");
+            PrintStream errorPrintStream = new PrintStream(buildLog);
+            logger.setErrorPrintStream(errorPrintStream);
+            logger.setOutputPrintStream(System.out);
+            logger.setMessageOutputLevel(Project.MSG_ERR);
+            buildProject.addBuildListener(logger);
+            
+            try {
+                
+                buildProject.fireBuildStarted();
+                ProjectHelper helper = ProjectHelper.getProjectHelper();
+                buildProject.addReference("ant.projectHelper", helper);
+                helper.parse(buildProject, buildFile);
+                buildProject.executeTarget("compile-test");
+                buildProject.fireBuildFinished(null);
+                return true;
+                
+            } catch (BuildException e) {
+                
+                buildProject.fireBuildFinished(e);
+                buildRunResult = new RunResult(Status.COMPILE_FAILED, ImmutableList.copyOf(new ArrayList<TestResult>()),
+                        new ImmutableMap.Builder<String, byte[]>().put(SpecialLogs.COMPILER_OUTPUT,
+                                java.nio.file.Files.readAllBytes(buildLog.toPath())).build());
+                return false;
+                
+            }
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private List<String> buildTestScannerArgs(Path path, ClassPath classPath, String... args) {
+        List<String> scannerArgs = new ArrayList<>();
+
+        scannerArgs.add("java");
+        scannerArgs.add("-cp");
+
+        if (classPath == null) {
+            scannerArgs.add(generateClassPath(path).toString());
+        } else {
+            scannerArgs.add(classPath.toString());
+        }
+
+        scannerArgs.add("fi.helsinki.cs.tmc.testscanner.TestScanner");
+        path = path.toAbsolutePath();
+        scannerArgs.add(path.toString() + testDir);
+
+        if (args != null) {
+            for (String arg : args) {
+                scannerArgs.add(arg);
+            }
+        }
```

Comment:
Most likely it would, but once test-scanner is brought to tmc-langs buildTestScannerArgs is redundant and we can remove the whole method.